Argument missing for parameter

There's a bug with the type hinting in version 3.7.0+

```python
from asgiref.sync import sync_to_async


class Whatever:
    @sync_to_async()
    def something(self, i: int) -> int:
        return i + 1


what = Whatever()


async def main() -> None:
    await what.something(i=1)  # error is here
```

Produces the following error in `pyright`:

> Argument missing for parameter "self" [reportGeneralTypeIssues]

And with `mypy`:

> error: Too few arguments  [call-arg]

This code did not produce any type hinting issues in version 3.6.0 and prior.

- pyright 1.1.309
- mypy 1.30
- python 3.10.10

--- a/asgiref/sync.py
+++ b/asgiref/sync.py
@@ -554,8 +554,54 @@ class SyncToAsync(Generic[_P, _R]):
             return None
 
 
-# Lowercase aliases (and decorator friendliness)
-async_to_sync = AsyncToSync
+@overload
+def async_to_sync(
+    *,
+    force_new_loop: bool = False,
+) -> Callable[
+    [Union[Callable[_P, Coroutine[Any, Any, _R]], Callable[_P, Awaitable[_R]]]],
+    Callable[_P, _R],
+]:
+    ...
+
+
+@overload
+def async_to_sync(
+    awaitable: Union[
+        Callable[_P, Coroutine[Any, Any, _R]],
+        Callable[_P, Awaitable[_R]],
+    ],
+    *,
+    force_new_loop: bool = False,
+) -> Callable[_P, _R]:
+    ...
+
+
+def async_to_sync(
+    awaitable: Optional[
+        Union[
+            Callable[_P, Coroutine[Any, Any, _R]],
+            Callable[_P, Awaitable[_R]],
+        ]
+    ] = None,
+    *,
+    force_new_loop: bool = False,
+) -> Union[
+    Callable[
+        [Union[Callable[_P, Coroutine[Any, Any, _R]], Callable[_P, Awaitable[_R]]]],
+        Callable[_P, _R],
+    ],
+    Callable[_P, _R],
+]:
+    if awaitable is None:
+        return lambda f: AsyncToSync(
+            f,
+            force_new_loop=force_new_loop,
+        )
+    return AsyncToSync(
+        awaitable,
+        force_new_loop=force_new_loop,
+    )
 
 
 @overload
@@ -563,7 +609,7 @@ def sync_to_async(
     *,
     thread_sensitive: bool = True,
     executor: Optional["ThreadPoolExecutor"] = None,
-) -> Callable[[Callable[_P, _R]], SyncToAsync[_P, _R]]:
+) -> Callable[[Callable[_P, _R]], Callable[_P, Coroutine[Any, Any, _R]]]:
     ...
 
 
@@ -573,15 +619,19 @@ def sync_to_async(
     *,
     thread_sensitive: bool = True,
     executor: Optional["ThreadPoolExecutor"] = None,
-) -> SyncToAsync[_P, _R]:
+) -> Callable[_P, Coroutine[Any, Any, _R]]:
     ...
 
 
 def sync_to_async(
     func: Optional[Callable[_P, _R]] = None,
+    *,
     thread_sensitive: bool = True,
     executor: Optional["ThreadPoolExecutor"] = None,
-) -> Union[Callable[[Callable[_P, _R]], SyncToAsync[_P, _R]], SyncToAsync[_P, _R]]:
+) -> Union[
+    Callable[[Callable[_P, _R]], Callable[_P, Coroutine[Any, Any, _R]]],
+    Callable[_P, Coroutine[Any, Any, _R]],
+]:
     if func is None:
         return lambda f: SyncToAsync(
             f,

